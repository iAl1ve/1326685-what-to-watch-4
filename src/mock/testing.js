export const ListMovies = [
  {
    id: 1,
    title: `The Grand Budapest Hotel`,
    genre: `Comedies`,
    year: 1996,
    src: `img/the-grand-budapest-hotel-poster.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 8.9,
    ratingCount: 256,
    movieDescription: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
    Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    movieDirector: `Wes Andreson`,
    movieStarring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: 122,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    backgroundColor: `#B41D8F`,
    previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  }, {
    id: 2,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Drama`,
    year: 2013,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 9.1,
    ratingCount: 146,
    movieDescription: `Fantastic Beasts: The Crimes of Grindelwald movie description text. Fantastic Beasts: The Crimes of Grindelwald movie description text. Fantastic Beasts: The Crimes of Grindelwald movie description text. `,
    movieDirector: `Wes Andreson`,
    movieStarring: [`Bill Murray`, `Willem Dafoe and other`],
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: 131,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    backgroundColor: `#B41D81`,
    previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  }, {
    id: 3,
    title: `Bohemian Rhapsody`,
    src: `img/bohemian-rhapsody.jpg`,
    genre: `Comedies`,
    year: 2015,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 4.1,
    ratingCount: 195,
    movieDescription: `Bohemian Rhapsody movie description text. Bohemian Rhapsody movie description text. Bohemian Rhapsody movie description text. `,
    movieDirector: `Bill Murray`,
    movieStarring: [`Edward Norton`, `Jude Law`, `Willem Dafoe`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 103,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    backgroundColor: `#B31D8F`,
    previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  }, {
    id: 4,
    title: `Macbeth`,
    src: `img/macbeth.jpg`,
    genre: `Drama`,
    year: 2014,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 9.3,
    ratingCount: 119,
    movieDescription: `Macbeth movie description text. Macbeth movie description text. Macbeth movie description text. Macbeth movie description text. `,
    movieDirector: `Macbeth`,
    movieStarring: [`Macbeth`, `Willem Dafoe`],
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: 212,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    backgroundColor: `#B40D8F`,
    previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  }, {
    id: 5,
    title: `Aviator`,
    src: `img/aviator.jpg`,
    genre: `Documentary`,
    year: 2013,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 9.2,
    ratingCount: 249,
    movieDescription: `Aviator movie description text. Aviator movie description text. Aviator movie description text. Aviator movie description text.`,
    movieDirector: `Aviator`,
    movieStarring: [`Bill Murray`, `Aviator`, `Willem Dafoe`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 165,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    backgroundColor: `#A31D8F`,
    previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  }, {
    id: 6,
    title: `Moonrise Kingdom`,
    src: `img/moonrise-kingdom.jpg`,
    genre: `Romance`,
    year: 2011,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 9.1,
    ratingCount: 232,
    movieDescription: `Moonrise Kingdom movie description text. Moonrise Kingdom movie description text. Moonrise Kingdom movie description text. Moonrise Kingdom movie description text.`,
    movieDirector: `Moonrise Kingdom`,
    movieStarring: [`Moonrise Kingdom`, `Jude Law`],
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: 138,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    backgroundColor: `#C51D8F`,
    previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  }, {
    id: 7,
    title: `Midnight Special`,
    src: `img/midnight-special.jpg`,
    genre: `Crime`,
    year: 2016,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 8.4,
    ratingCount: 212,
    movieDescription: `Midnight Special movie description text. Midnight Special movie description text. Midnight Special movie description text. Midnight Special movie description text.`,
    movieDirector: `Midnight Special`,
    movieStarring: [`Midnight Special`, `Jude Law`, `Willem Dafoe`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 154,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    backgroundColor: `#D31D8F`,
    previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  }, {
    id: 8,
    title: `War of the Worlds`,
    src: `img/war-of-the-worlds.jpg`,
    genre: `Drama`,
    year: 2013,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 8.9,
    ratingCount: 201,
    movieDescription: `War of the Worlds movie description text. War of the Worlds movie description text. War of the Worlds movie description text. War of the Worlds movie description text. War of the Worlds movie description text. `,
    movieDirector: `War of the Worlds`,
    movieStarring: [`War of the Worlds`],
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: 141,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    backgroundColor: `#B40A8F`,
    previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  }, {
    id: 9,
    title: `Shutter Island`,
    src: `img/shutter-island.jpg`,
    genre: `Horror`,
    year: 2017,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 7.9,
    ratingCount: 175,
    movieDescription: `Shutter Island movie description text. Shutter Island movie description text. Shutter Island movie description text. Shutter Island movie description text. `,
    movieDirector: `Shutter Island`,
    movieStarring: [`Bill Murray`, `Shutter Island`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 109,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    backgroundColor: `#A61D8F`,
    previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  }
];

export const ListReviews = [
  {
    id: 1,
    idUser: 2,
    text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    author: `Kate Muir`,
    date: `2016-12-24`,
    rating: 7.6,
  }, {
    id: 2,
    idUser: 2,
    text: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    author: `Matthew Lickona`,
    date: `2016-12-20`,
    rating: 8.3,
  }, {
    id: 3,
    idUser: 2,
    text: `Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.`,
    author: `Bill Goodykoontz`,
    date: `2015-11-18`,
    rating: 7.3,
  },
  {
    id: 4,
    idUser: 3,
    text: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    author: `Paula Fleri-Soler`,
    date: `2016-12-20`,
    rating: 7.8,
  },
  {
    id: 5,
    idUser: 1,
    text: `When one of Gustave's lovers dies mysteriously. It is certainly a magical and childlike way of storytelling.`,
    author: `Amanda Soler`,
    date: `2016-12-20`,
    rating: 9.6,
  },
  {
    id: 6,
    idUser: 3,
    text: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
    author: `Amanda Greever`,
    date: `2015-11-18`,
    rating: 8.9,
  },
  {
    id: 7,
    idUser: 4,
    text: `Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    author: `Bill Muir`,
    date: `2016-08-14`,
    rating: 8.4,
  },
  {
    id: 8,
    idUser: 5,
    text: `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs.`,
    author: `Wes  Dafoe`,
    date: `2015-11-21`,
    rating: 5.6,
  },
  {
    id: 9,
    idUser: 2,
    text: `Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    author: `Kate Norton`,
    date: `2015-03-08`,
    rating: 2,
  },
  {
    id: 10,
    idUser: 1,
    text: `Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    author: `Andreson Fleri`,
    date: `2016-05-09`,
    rating: 10,
  },
];

export const listGenres = [`All genres`, `Comedies`, `Drama`, `Documentary`, `Romance`, `Crime`, `Horror`];

export const notAdaptedListReviews = [
  {
    id: 1,
    user: {
      id: 2,
      name: `Kate Muir`
    },
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2016-12-24`,
    rating: 7.6,
  }
];

export const adaptedUserInfo = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatarUrl: `img/1.png`
};
