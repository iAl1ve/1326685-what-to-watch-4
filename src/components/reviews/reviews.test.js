import React from "react";
import renderer from "react-test-renderer";
import Reviews from "./reviews.jsx";
import {ListReviews} from "../../mock/testing.js";

describe(`Test Render Reviews`, () => {
  it(`Render Reviews`, () => {
    const tree = renderer
      .create(<Reviews
        reviews = {ListReviews}
      />, {
        createNodeMock: () => {
          return {};
        }
      }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
