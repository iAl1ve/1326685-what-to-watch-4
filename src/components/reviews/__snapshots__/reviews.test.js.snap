// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Render Reviews Render Reviews 1`] = `
<div
  className="movie-card__reviews-col"
>
  <div
    className="review"
  >
    <blockquote
      className="review__quote"
    >
      <p
        className="review__text"
      >
        Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
      </p>
      <footer
        className="review__details"
      >
        <cite
          className="review__author"
        >
          Kate Muir
        </cite>
        <time
          className="review__date"
          dateTime="{review.date}"
        >
          December
    24,
    2016
        </time>
      </footer>
    </blockquote>
    <div
      className="review__rating"
    >
      7.6
    </div>
  </div>
  <div
    className="review"
  >
    <blockquote
      className="review__quote"
    >
      <p
        className="review__text"
      >
        The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.
      </p>
      <footer
        className="review__details"
      >
        <cite
          className="review__author"
        >
          Matthew Lickona
        </cite>
        <time
          className="review__date"
          dateTime="{review.date}"
        >
          December
    20,
    2016
        </time>
      </footer>
    </blockquote>
    <div
      className="review__rating"
    >
      8.3
    </div>
  </div>
  <div
    className="review"
  >
    <blockquote
      className="review__quote"
    >
      <p
        className="review__text"
      >
        Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.
      </p>
      <footer
        className="review__details"
      >
        <cite
          className="review__author"
        >
          Bill Goodykoontz
        </cite>
        <time
          className="review__date"
          dateTime="{review.date}"
        >
          November
    18,
    2015
        </time>
      </footer>
    </blockquote>
    <div
      className="review__rating"
    >
      7.3
    </div>
  </div>
  <div
    className="review"
  >
    <blockquote
      className="review__quote"
    >
      <p
        className="review__text"
      >
        It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.
      </p>
      <footer
        className="review__details"
      >
        <cite
          className="review__author"
        >
          Paula Fleri-Soler
        </cite>
        <time
          className="review__date"
          dateTime="{review.date}"
        >
          December
    20,
    2016
        </time>
      </footer>
    </blockquote>
    <div
      className="review__rating"
    >
      7.8
    </div>
  </div>
  <div
    className="review"
  >
    <blockquote
      className="review__quote"
    >
      <p
        className="review__text"
      >
        When one of Gustave's lovers dies mysteriously. It is certainly a magical and childlike way of storytelling.
      </p>
      <footer
        className="review__details"
      >
        <cite
          className="review__author"
        >
          Amanda Soler
        </cite>
        <time
          className="review__date"
          dateTime="{review.date}"
        >
          December
    20,
    2016
        </time>
      </footer>
    </blockquote>
    <div
      className="review__rating"
    >
      9.6
    </div>
  </div>
  <div
    className="review"
  >
    <blockquote
      className="review__quote"
    >
      <p
        className="review__text"
      >
        I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.
      </p>
      <footer
        className="review__details"
      >
        <cite
          className="review__author"
        >
          Amanda Greever
        </cite>
        <time
          className="review__date"
          dateTime="{review.date}"
        >
          November
    18,
    2015
        </time>
      </footer>
    </blockquote>
    <div
      className="review__rating"
    >
      8.9
    </div>
  </div>
  <div
    className="review"
  >
    <blockquote
      className="review__quote"
    >
      <p
        className="review__text"
      >
        Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
      </p>
      <footer
        className="review__details"
      >
        <cite
          className="review__author"
        >
          Bill Muir
        </cite>
        <time
          className="review__date"
          dateTime="{review.date}"
        >
          August
    14,
    2016
        </time>
      </footer>
    </blockquote>
    <div
      className="review__rating"
    >
      8.4
    </div>
  </div>
  <div
    className="review"
  >
    <blockquote
      className="review__quote"
    >
      <p
        className="review__text"
      >
        Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs.
      </p>
      <footer
        className="review__details"
      >
        <cite
          className="review__author"
        >
          Wes  Dafoe
        </cite>
        <time
          className="review__date"
          dateTime="{review.date}"
        >
          November
    21,
    2015
        </time>
      </footer>
    </blockquote>
    <div
      className="review__rating"
    >
      5.6
    </div>
  </div>
  <div
    className="review"
  >
    <blockquote
      className="review__quote"
    >
      <p
        className="review__text"
      >
        Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.
      </p>
      <footer
        className="review__details"
      >
        <cite
          className="review__author"
        >
          Kate Norton
        </cite>
        <time
          className="review__date"
          dateTime="{review.date}"
        >
          March
    8,
    2015
        </time>
      </footer>
    </blockquote>
    <div
      className="review__rating"
    >
      2
    </div>
  </div>
  <div
    className="review"
  >
    <blockquote
      className="review__quote"
    >
      <p
        className="review__text"
      >
        Zero, a junior lobby boy, becomes Gustave's friend and protege.
      </p>
      <footer
        className="review__details"
      >
        <cite
          className="review__author"
        >
          Andreson Fleri
        </cite>
        <time
          className="review__date"
          dateTime="{review.date}"
        >
          May
    9,
    2016
        </time>
      </footer>
    </blockquote>
    <div
      className="review__rating"
    >
      10
    </div>
  </div>
</div>
`;
